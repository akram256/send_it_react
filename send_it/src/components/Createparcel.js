import React, { Component } from 'react'
import createparcel from '../actions/createparcelActions'
import { connect } from 'react-redux'
import {Link} from 'react-router-dom'

export class Createparcel extends Component {
    constructor(props) {
        super(props);
        this.state = {
            parcel_name: '',
            pickup_location: '',
            destination: '',
            reciever: '',
            weight: ''
        };
        this.onChange = this.onChange.bind(this);
        this.onSubmit = this.onSubmit.bind(this);
    }
    onChange(e) {
        this.setState({ [e.target.name]: e.target.value });
    }
    onSubmit(e) {
        e.preventDefault();
        this.props.createparcel({
            parcel_name: this.state.parcel_name,
            pickup_location: this.state.pickup_location, destination: this.state.destination,
            reciever: this.state.reciever, weight: parseInt(this.state.weight)
        }, this.props)
    }
    render() {
        return (
            <div>
                <nav class="navbar navbar-expand-lg navbar-light bg-light">
                    <div className="container">
                        <a class="navbar-brand" href="#">SEND_IT</a>
                        <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
                            <span class="navbar-toggler-icon"></span>
                        </button>

                        <div class="collapse navbar-collapse" id="navbarSupportedContent">
                            <ul class="navbar-nav ml-auto">
                                <li class="nav-item">

                                    <Link to='/view_parcels' class="nav-link" >View Parcel Orders</Link>
                                </li>
                                <li class="nav-item">

                                    <a href="profile.html" class="nav-link">Profile</a>
                                </li>
                                <li class="nav-item">
                                    <a href="index.html" class="nav-link">(Logout)</a>

                                </li>


                            </ul>

                        </div>
                    </div>
                </nav>
                <div class="content">
                    <div class="orders">
                        <h4>Welcome to SEND_IT.</h4>
                        <h6><i>"Have it, Send it faster "</i></h6>

                        <div class="menu">
                            LET'S SEND IT NOW
                    </div>
                        <form className="order-item" onSubmit={this.onSubmit}>
                            <div className="order-content">
                                <div class="title">Parcel Gift  </div>
                                <div class="field">
                                    <input type="text" name="parcel_name"
                                        value={this.state.parcel_name} onChange={this.onChange} placeholder="parcel_name" required />
                                    <input type="text" name="pickup_location"
                                        value={this.state.pickup_location} onChange={this.onChange} placeholder="pickup_location" required />
                                    <input type="text" name="destination"
                                        value={this.state.destination} onChange={this.onChange} placeholder="destination" required />
                                    <input type="text" name="reciever"
                                        value={this.state.reciever} onChange={this.onChange} placeholder="reciever" required />
                                    <input type="number" name="weight"
                                        value={this.state.weight} onChange={this.onChange} placeholder="weight" required />

                                </div>

                            </div>
                            <div className="order-button">
                                <button className="button completed">Create_order</button>
                            </div>

                        </form>
                    </div>
                </div>
            </div>
        )
    }
}
export const mapStateToProps = state => ({
    createdata: state.createReducers.createdata,
    errors: state.createReducers.errors,
    isSuccessfully_created: state.createReducers.isSuccessfully_created,
});


export default connect(mapStateToProps, { createparcel })(Createparcel);
