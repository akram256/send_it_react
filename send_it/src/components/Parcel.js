import React, { Component } from 'react'

export const Parcel = props => {
    return (
      <tr>
        <td>{props.parcel_id}</td>
        <td>{props.parcel_name}</td>
        <td>{props.pickup_location}</td>
        <td>{props.destination}</td>
        <td>{props.reciever}</td>
        <td>{props.weight}</td>
        <td>{props.parcel_status}</td>
        <td>{props.order_date}</td>
        <td>{props.current_location}</td>
      </tr>
    );
  };
  
  export default Parcel;