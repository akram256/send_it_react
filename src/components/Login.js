import React, { Component } from 'react'
import '../styles/login.css'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'
import loginuser from '../actions/loginActions'

export class Login extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: ''
    };
    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }
  onChange(e) {
    this.setState({ [e.target.name]: e.target.value });
  }
  onSubmit(e) {
    e.preventDefault();
    this.props.loginuser({ email: this.state.email, password: this.state.password }, this.props)
  }
  render() {
    return (
      <div>
        <nav class="navbar navbar-expand-lg navbar-light bg-light">
          <div className="container">
            <a class="navbar-brand" href="#">SEND_IT</a>
            <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
              <span class="navbar-toggler-icon"></span>
            </button>

            <div class="collapse navbar-collapse" id="navbarSupportedContent">
              <ul class="navbar-nav ml-auto">
                <li class="nav-item">

                  <Link to='/' class="nav-link" >Login</Link>
                </li>
                <li class="nav-item">
                  <Link to='/register' className="nav-link">signup</Link>

                </li>
   
              </ul>

            </div>
          </div>
        </nav>


        <div className="content">
          <div className="form">
            <form className="signup" onSubmit={this.onSubmit}>
              <h4>create an account</h4>
              <div className="field">
                <label >email</label>
                <input type="email" name="email" value={this.state.email} onChange={this.onChange} required />
              </div>
              <div className="field">
                <label >Password</label>
                <input type="password" name="password" value={this.state.password} onChange={this.onChange} min="8" required />
              </div>
              <div className="button">
                <button type="submit" className="submit">Login</button>
              </div>
              <br />

              <div><Link to='/register' className="link">Don't Have an account? SignUp here.</Link></div>

            </form>

          </div>
        </div>
      </div>
    )
  }
}
export const mapStateToProps = state => ({
  token: state.loginReducers.token,
  errors: state.loginReducers.errors,
  isSuccessful: state.loginReducers.isSuccessful,
});

export default connect(mapStateToProps, { loginuser })(Login)
