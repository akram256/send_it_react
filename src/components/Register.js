import React, { Component } from 'react'
import '../styles/login.css'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'
import registeruser from '../actions/registerActions'

export class Register extends Component {
    constructor(props) {
        super(props);
        this.state = {
            user_name: '',
            email: '',
            password: ''
        };
        this.onChange = this.onChange.bind(this);
        this.onSubmit = this.onSubmit.bind(this);
    }
    onChange(e) {
        this.setState({ [e.target.name]: e.target.value });
    }
    onSubmit(e) {
        e.preventDefault();
        this.props.registeruser({ user_name: this.state.user_name, email: this.state.email, password: this.state.password }, this.props)
    }

    render() {
        return (

            <div>
                <nav class="navbar navbar-expand-lg navbar-light bg-light">
                    <div className="container">
                        <a class="navbar-brand" href="#">SEND_IT</a>
                        <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
                            <span class="navbar-toggler-icon"></span>
                        </button>

                        <div class="collapse navbar-collapse" id="navbarSupportedContent">
                            <ul class="navbar-nav ml-auto">
                                <li class="nav-item">

                                    <Link to='/' class="nav-link" >Login</Link>
                                </li>
                                <li class="nav-item">
                                    <Link to='/register' className="nav-link">signup</Link>

                                </li>

                            </ul>

                        </div>
                    </div>
                </nav>
                <div className="content">
                    <div className="form">
                        <form className="signup" onSubmit={this.onSubmit}>
                            <h4>create an account</h4>

                            <div className="field">
                                <label for="username">username</label>
                                <input type="text" name="user_name" value={this.state.user_name} onChange={this.onChange} required />
                            </div>
                            <div class="field">
                                <label for="email">email</label>
                                <input type="email" name="email" value={this.state.email} onChange={this.onChange} required />
                            </div>
                            <div class="field">
                                <label for="password">Password</label>
                                <input type="password" name="password" value={this.state.password} onChange={this.onChange} min="8" required />
                            </div>
                            <div class="button">
                                <button type="submit" value="submit" className="submit">SignUp</button>
                            </div>
                            <br />
                            <div><Link to='/' className="link">Have an account? Login here.</Link></div>

                        </form>

                    </div>
                </div>
            </div>
        )
    }
}
export const mapStateToProps = state => ({
    token: state.registerReducers.token,
    errors: state.registerReducers.errors,
    isSuccessful: state.registerReducers.isSuccessful,
});

export default connect(mapStateToProps, { registeruser })(Register)
