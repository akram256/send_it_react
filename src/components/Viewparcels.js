import React, { Component } from 'react'
import AllParcels from './ParcelTable'
import getallparcels from '../actions/getparcelsAction'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'

export class Viewparcels extends Component {
    constructor(props) {
        super(props);
        this.state = {

        };
    }

    componentDidMount() {

        this.props.getallparcels();
    }


    render() {


        return (
            <div>
                <nav class="navbar navbar-expand-lg navbar-light bg-light">
                    <div className="container">
                        <a class="navbar-brand" href="#">SEND_IT</a>
                        <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
                            <span class="navbar-toggler-icon"></span>
                        </button>

                        <div class="collapse navbar-collapse" id="navbarSupportedContent">
                            <ul class="navbar-nav ml-auto">
                                <li class="nav-item">

                                    <Link to='/create_parcel' class="nav-link" >Create Parcel Order</Link>
                                </li>
                                <li class="nav-item">

                                    <a href="profile.html" class="nav-link">Profile</a>
                                </li>
                                <li class="nav-item">
                                    <a href="index.html" class="nav-link">(Logout)</a>

                                </li>


                            </ul>

                        </div>
                    </div>
                </nav>

                <div class="content mt-4">
                    <div class="orders">
                        <h4>All Parcel Delivery History.</h4>

                        <div class="menu">
                            VIEW ALL YOUR PARCEL ORDER
                        </div>
                        <div class="order-item">
                            <div class="order-content">
                                <div id="specificparcels_table">
                                    <AllParcels parcels={this.props.parcels} />


                                </div>

                            </div>
                        </div>
                    </div>
                </div>
            </div>
        )
    }
}
export const mapStateToProps = state => ({
    parcels: state.parcels.parcels,
    errors: state.parcels.errors,
    isSuccessfully_retrived: state.parcels.isSuccessfully_retrived,
});


export default connect(mapStateToProps, { getallparcels })(Viewparcels);