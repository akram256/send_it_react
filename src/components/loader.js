import React, { Component } from 'react'

export class Loader extends Component {
    render() {
        return (
            <div>
                <div class="spinner-grow" role="status">
                    <span class="sr-only">Loading...</span>
                </div>
            </div>
        )
    }
}

export default Loader
