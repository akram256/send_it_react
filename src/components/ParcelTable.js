import React from "react";
import Parcel from "./Parcel";

export const AllParcels = props => {
  const retriveddata = props.parcels ? props.parcels : [];
  const allparcels = retriveddata.map(parcel => (
    <Parcel
      key={parcel.parcel_id}
      parcel_id={parcel.parcel_id}
      parcel_name={parcel.parcel_name}
      destination={parcel.destination}
      pickup_location={parcel.pickup_location}
      reciever={parcel.reciever}
      weight={parcel.weight}
      current_location={parcel.current_location}
      order_date={parcel.order_date}
      parcel_status={parcel.parcel_status}
    />
  ));

  return (
    <div>
     
      <table className="table table-bordered table-dark">
        <thead className="thead-dark">
          <tr>
            <th>Parcel_Id</th>
            <th>parcel_name</th>
            <th>destination</th>
            <th>Pickup_location</th>
            <th>reciever</th>
            <th>weight</th>
            <th>status</th>
            <th>order_date</th>
            <th>current_location</th>
          
            <th />
          </tr>
        </thead>
        <tbody>{allparcels}</tbody>
      </table>
    </div>
  );
};

export default AllParcels;
