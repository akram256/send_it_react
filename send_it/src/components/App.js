import React, { Component } from 'react'
import Login from './Login'
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import Register from './Register'
import { createBrowserHistory } from 'history';
import Createparcel from './Createparcel';
import Viewparcels from './Viewparcels'
class App extends Component {

  render() {

    return (
      <div>
        <Router history={createBrowserHistory}>
          <Switch>
            <Route exact path="/" component={Login} />
            <Route exact path="/register" component={Register} />
            <Route exact path="/create_parcel" component={Createparcel} />
            <Route exact path="/view_parcels" component={Viewparcels} />
            <Login />
            <Register />
          </Switch>
        </Router>
      </div>
    )
  }
}

export default App
