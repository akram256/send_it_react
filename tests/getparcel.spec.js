import React from 'react';
import { shallow, mount } from 'enzyme';
import {Viewparcels} from '../src/components/Viewparcels';

describe('get parcel component', () => {
  const  props = {
     getallparcels: jest.fn(),
   
  }
it('should render without crashing', () => {
  const wrapper = shallow(
  <Viewparcels {...props}/>
  );
  expect(wrapper).toMatchSnapshot();
});


});