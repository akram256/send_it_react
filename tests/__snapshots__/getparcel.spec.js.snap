// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get parcel component should render without crashing 1`] = `ShallowWrapper {}`;
